handler/enqueue_annotation.go

```diff
@@ -80,13 +136,14 @@ func (e *EnqueueRequestForAnnotation) Generic(evt event.GenericEvent, q workqueu
 	}
 }
 
+// getAnnotationRequests will check if the object has the annotations for the watch handler and requeue.
 func (e *EnqueueRequestForAnnotation) getAnnotationRequests(object metav1.Object) (bool, reconcile.Request) {
-	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type {
+	if typeString, ok := object.GetAnnotations()[TypeAnnotation]; ok && typeString == e.Type.String() {
 		namespacedNameString, ok := object.GetAnnotations()[NamespacedNameAnnotation]
 		if !ok {
 			log.Info("Unable to find namespaced name annotation for resource", "resource", object)
 		}
-		if namespacedNameString == "" {
+		if len(namespacedNameString) < 1 {
```

Comment:
I'm not really a fan of this empty string comparison operation. `str == ""` is much clearer. Ditto below.

Reply:
@estroz strings can be "<space><space><space> " so the cooperation == "" is pitiful.

Reply:
You can do `strings.TrimSpace(namespacedNameString) == ""` if you really care about space errors in user input. I don't expect there to ever be spaces in a GVK (below) since parsing such a GVK in the apiserver will result in errors.